### PR TITLE
Add Dockerfile, Makefile and use `uv` instead of `poetry`

### PR DESCRIPTION
**Pull request by rnetser**
- Move to use [uv](https://docs.astral.sh/uv/) 
- Add Dockerfile and Makefile

### CODE REVIEW

**Review by adolfo-ab — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    && tar xvf /tmp/rosa-linux.tar.gz --no-same-owner \
+    && mv rosa /usr/bin/rosa \
+    && chmod +x /usr/bin/rosa \
+    && rosa version
+
+# Install the OpenShift CLI (OC)
+RUN curl -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable/openshift-client-linux.tar.gz --output /tmp/openshift-client-linux.tar.gz \
+    && tar xvf /tmp/openshift-client-linux.tar.gz --no-same-owner \
+    && mv oc /usr/bin/oc \
+    && mv kubectl /usr/bin/kubectl \
```
Comment (mwaykole) [inline Dockerfile:26]: why kubectl ?

File: pyproject.toml [hunk 2]
```diff
@@ -16,42 +16,34 @@ no_implicit_optional = true
...
+]
+
+[project]
+requires-python = ">=3.8"
 name = "opendatahub-tests"
 version = "0.1.0"
 description = "Tests repository for Open Data Hub (ODH)"
 authors = []
-license = "Apache License 2.0"
+license = "Apache-2.0"
```
Comment (mwaykole) [inline pyproject.toml:31]: are u sure aboout `License`

**Review by mwaykole — DISMISSED** [review]
few comments else looks good

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+IMAGE_BUILD_CMD = $(shell which podman 2>/dev/null || which docker)
+IMAGE_REGISTRY ?= "quay.io"
+REGISTRY_NAMESPACE ?= "modh"
+IMAGE_NAME="opendatahub-tests"
+IMAGE_TAG ?= "latest"
+
+FULL_OPERATOR_IMAGE ?= "$(IMAGE_REGISTRY)/$(REGISTRY_NAMESPACE)/$(IMAGE_NAME):$(IMAGE_TAG)"
+POETRY_BIN = poetry
```
Comment (lugi0) [inline Makefile:8]: I think this is not needed if we switch to uv?

File: pyproject.toml (hunk 2 above)
Comment (lugi0) [reply pyproject.toml:31]: I vote for agpl 3

**Review by lugi0 — DISMISSED** [review]
I have my concerns about switching to `uv` since it's a newcomer and written in Rust, but we can evaluate it and switch back to Poetry if needed

File: pyproject.toml (hunk 2 above)
Comment (rnetser) [reply pyproject.toml:31]: why not apache? checking `kserve` and `opendatahub-operator` as examples of repos in the org - they use apache. we should use the same

File: Makefile (hunk 3 above)
Comment (rnetser) [reply Makefile:8]: leftover, removing

File: Dockerfile (hunk 1 above)
Comment (rnetser) [reply Dockerfile:26]: will remove

File: pyproject.toml (hunk 2 above)
Comment (rnetser) [reply pyproject.toml]: > are u sure aboout `License`

updated the string

**Review by lugi0 — APPROVED** [review]

**Review by mwaykole — APPROVED** [review]

**Pull request merged**